src/components/TextInput/TextInput.tsx

```diff
@@ -53,6 +68,24 @@ export const TextInput: React.FC<TextInputProps> = ({
 
   const inputRef = useRef<HTMLInputElement>(null);
 
+  const [eyeIcon, setEyeIcon] = useState("view-on");
+  const [inputType, setInputType] = useState(type);
```

Comment:
@thota499, PS, related to why the type wasn't changing when you adjusted it via the Story Controls, it's because you set the `type` here, but never update the `inputType` if the user changes the `type` prop. Which I think is fine. If you _wanted_ to handle that scenario, you'd want to use an effect like so (but again, this all looks good to me as-is):
```suggestion
  const [inputType, setInputType] = useState(type);
  useEffect(() => setInputType(type), [type]);
```